archinstall/lib/disk.py

```diff
@@ -518,12 +523,20 @@ def use_entire_disk(self, root_filesystem_type='ext4'):
 			self.blockdevice.partition[0].allow_formatting = True
 			self.blockdevice.partition[1].allow_formatting = True
 		else:
-			# we don't need a seprate boot partition it would be a waste of space
-			self.add_partition('primary', start='1MB', end='100%')
-			self.blockdevice.partition[0].filesystem = root_filesystem_type
-			log(f"Set the root partition {self.blockdevice.partition[0]} to use filesystem {root_filesystem_type}.", level=logging.DEBUG)
-			self.blockdevice.partition[0].target_mountpoint = '/'
+			self.add_partition('primary', start='1MiB', end='513MiB', partition_format='ext2')
+			self.set(0, 'boot on')
+			self.add_partition('primary', start='513MiB', end='100%')
+
+			self.blockdevice.partition[0].filesystem = 'ext2'
```

Comment:
Is `ext2` a hard dependency to get this working on GRUB? Or can we retaim `.filesystem = root_filesystem_type`?

Reply:
> Is `ext2` a hard dependency to get this working on GRUB? Or can we retaim `.filesystem = root_filesystem_type`?

Most articles I found about boot w/ grub mention that the format should be either ext2, ext3, or ext4. I am not sure tbh which one is better, however, I wouldn't leave the format related to users choice for root since xfs, btrfs, or other formats may not work with grub in separate boot.

https://unix.stackexchange.com/questions/175728/what-file-system-should-a-grub-2-boot-partition-use

Thanks

Reply:
In that case, I think `ext4` would be better if we're going to hard code anything. Which means the user's choice of root partition filesystem will be ignored. @advaithm can probably chime in here :)

Reply:
There isn't a hard dep on the file system however if we do chose to hard code fs etx4 would be the best choice

Reply:
note for btrfs on mbr you may have to offset the start partition by 1mb

Reply:
Never had issues with ext4, speaking of /boot partition though keep in mind that luks2 support for grub2 is [limited](https://www.gnu.org/software/grub/manual/grub/html_node/cryptomount.html).